pkg/proto/aksnodeconfig/v1/tlsbootstrappingconfig.proto

```diff
@@ -1,13 +1,26 @@
 syntax = "proto3";
 package aksnodeconfig.v1;
- 
+
+enum BootstrappingAuthMethod  {
+  //nolint:gosec // this is a const string to use in switch statements, not hardcoded credentials
+  UseTLSBootstrapToken = 0;
+  UseSecureTLSBootstrapping = 1;
+  UseArcMsiToMakeCSR = 2;
+  UseAzureMsiToMakeCSR = 3;
+  UseArcMsiDirectly = 4;
+  UseAzureMsiDirectly = 5;
+}
+
 message TLSBootstrappingConfig {
   // Enable secure TLS bootstrapping for the node.
-  optional bool enable_secure_tls_bootstrapping = 1;
+  BootstrappingAuthMethod bootstrapping_auth_method = 1;
 
   // Only required until Secure TLS bootstrapping in place. Would use kubelet identity after that.
-  string tls_bootstrapping_token = 2; 
+  string tls_bootstrapping_token = 2;
```

Comment:
im not quite as familiar with proto best practices, but I think I read that marking each field as optional and then adding explicit comments to indicate required fields is generally considered best-practice

Reply:
note that we eventually plan on removing all references to bootstrap tokens / vanilla TLS bootstrapping once secure TLS bootstrapping has replaced it

Reply:
Have marked the 3 string fields as optional. Some Stretch clusters might want to use bootstrap tokens, so we prob shouldn't remove it entirely.